pyk/src/tests/integration/proof/test_imp.py

```diff
@@ -926,6 +944,66 @@ def test_terminal_node_subsumption(
             prover.advance_proof(proof, max_iterations=1)
             assert proof.status == ProofStatus.PASSED
 
+    @pytest.mark.parametrize(
+        'test_id,spec_file,spec_module,claim_id,max_iterations,max_depth,cut_rules,admit_deps,proof_status,expected_leaf_number',
+        APR_PROVE_TEST_DATA,
+        ids=[test_id for test_id, *_ in APR_PROVE_TEST_DATA],
+    )
+    def test_all_path_reachability_prove_parallel(
+        self,
+        kprove: KProve,
+        kcfg_explore: KCFGExplore,
+        test_id: str,
+        spec_file: str,
+        spec_module: str,
+        claim_id: str,
+        max_iterations: int | None,
+        max_depth: int | None,
+        cut_rules: Iterable[str],
+        admit_deps: bool,
+        proof_status: ProofStatus,
+        expected_leaf_number: int,
+        tmp_path_factory: TempPathFactory,
+        definition: KDefinition,
+        bug_report: BugReport,
+        kompiled_kore: KompiledKore,
+        _kore_server: KoreServer,
+    ) -> None:
+        with tmp_path_factory.mktemp(f'apr_tmp_proofs-{test_id}') as proof_dir:
+            spec_modules = kprove.get_claim_modules(Path(spec_file), spec_module_name=spec_module)
+            spec_label = f'{spec_module}.{claim_id}'
+            proofs = APRProof.from_spec_modules(
+                kprove.definition,
+                spec_modules,
+                spec_labels=[spec_label],
+                logs={},
+                proof_dir=proof_dir,
+            )
+            proof = single([p for p in proofs if p.id == spec_label])
+            if admit_deps:
+                for subproof in proof.subproofs:
+                    subproof.admit()
+                    subproof.write_proof_data()
+
+            def create_prover() -> APRProver:
+                kore_client = KoreClient(
+                    'localhost', _kore_server.port, timeout=self.CLIENT_TIMEOUT, bug_report=bug_report
+                )
+                ct_symb = CTermSymbolic(kore_client, definition, kompiled_kore)
+                _kcfg_explore = KCFGExplore(ct_symb, kcfg_semantics=self.semantics(definition))
```

Comment:
Consider defining a test class for parallel proving (similar to `KoreClientTest` et al.) that provides these building blocks as fixtures.